src/cares_wrap.cc

```diff
@@ -270,31 +285,43 @@ void ares_sockstate_cb(void* data,
 }
 
 
-Local<Array> HostentToAddresses(Environment* env, struct hostent* host) {
+Local<Array> HostentToAddresses(Environment* env,
+                                struct hostent* host,
+                                Local<Array> prev = Local<Array>()) {
   EscapableHandleScope scope(env->isolate());
-  Local<Array> addresses = Array::New(env->isolate());
+  auto context = env->context();
+  Context::Scope context_scope(context);
```

Comment:
In fact, I don't know very clearly. I just copied `Context::Scope` from other blocks of code.

eg.

```cpp
  void Parse(unsigned char* buf, int len) override {
    HandleScope handle_scope(env()->isolate());
    Context::Scope context_scope(env()->context());

    hostent* host;
    ares_addrttl addrttls[256];
    int naddrttls = arraysize(addrttls);

    int status = ares_parse_a_reply(buf, len, &host, addrttls, &naddrttls);
    if (status != ARES_SUCCESS) {
      ParseError(status);
      return;
    }

    Local<Array> addresses = HostentToAddresses(env(), host);
    Local<Array> ttls = Array::New(env()->isolate(), naddrttls);

    auto context = env()->context();
    for (int i = 0; i < naddrttls; i += 1) {
      auto value = Integer::New(env()->isolate(), addrttls[i].ttl);
      ttls->Set(context, i, value).FromJust();
    }
    ares_free_hostent(host);

    CallOnComplete(addresses, ttls);
  }
```